src/TestEnvironment.Docker/DockerEnvironment.cs

```diff
@@ -52,6 +56,74 @@ public void Dispose()
             }
         }
 
+        private async Task BuildRequiredImages(CancellationToken token)
+        {
+            foreach (var container in Dependencies.OfType<FromDockerfileContainer>())
+            {
+                // In order to pass the context we have to create tar file and use it as an argument.
+                var tempFileName = Guid.NewGuid().ToString();
+                using (var fileStream = new FileStream(tempFileName, FileMode.CreateNew))
+                {
+                    var tarArchive = TarArchive.CreateOutputTarArchive(fileStream);
```

Comment:
Maybe it would be useful to put tar archive creation-operations-deletion in try-catch-finally block?

Reply:
I prefer not to put too much try/catch in the code in order to avoid Exception hiding.